### PR TITLE
Bug 1548545 - add application-services signing workers.

### PR DESCRIPTION
**Pull request by MihaiTabara**
Not yet to be merged.

### CODE REVIEW

File: modules/signing_scriptworker/templates/dep-passwords-appsv.json.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "project:mozilla:application-services:releng:signing:cert:dep-signing": [
+        ["https://autograph-external.prod.autograph.services.mozaws.net", "signingscript_focus_dep", "<%= scope.function_secret(["autograph_focus_dep_password"]) %>", ["autograph_focus"], "autograph"]
```
Comment (MihaiTabara) [inline modules/signing_scriptworker/templates/dep-passwords-appsv.json.erb]: TODO: do we want different credentials here for Autograph?

File: modules/signing_scriptworker/templates/passwords-appsv.json.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "project:mozilla:application-services:releng:signing:cert:release-signing": [
```
Comment (MihaiTabara) [inline modules/signing_scriptworker/templates/passwords-appsv.json.erb]: TODO: do we want different credentials here for Autograph?

**Comment by MihaiTabara** [comment]
CC: @mitchhentges

File: modules/signing_scriptworker/manifests/settings.pp [hunk 3]
```diff
@@ -123,5 +123,35 @@
             datadog_api_key          => secret('scriptworker_datadog_api_key'),
             gpg_keyfile              => 'KEY_dep',
         },
+        'appsv-dep' => {
+            worker_type              => 'appsv-signing-dep-v1',
+            worker_group             => 'appsv-signing-dep-v1',
+            taskcluster_client_id    => 'project/mozilla/application-services/releng/scriptworker/signing/dep',
+            taskcluster_access_token => secret('application_services_signing_dep_scriptworker_taskcluster_access_token'),
+            passwords_template       => 'dep-passwords-appsv.json.erb',
+            scope_prefixes           => ['project:mobile:focus:releng:signing:', 'project:mobile:android-components:releng:signing:', 'project:mobile:fenix:releng:signing:', 'project:mobile:reference-browser:releng:signing:', 'project:mozilla:application-services:releng:signing'],
```
Comment (mitchhentges) [inline modules/signing_scriptworker/manifests/settings.pp]: We only need `project:mozilla:application-services:releng:signing` here, right?

File: modules/signing_scriptworker/manifests/settings.pp [hunk 4]
```diff
@@ -123,5 +123,35 @@
...
+            taskcluster_access_token => secret('application_services_signing_scriptworker_taskcluster_access_token'),
+            passwords_template       => 'passwords-appsv.json.erb',
+            scope_prefixes           => ['project:mobile:focus:releng:signing:', 'project:mobile:android-components:releng:signing:', 'project:mobile:fenix:releng:signing:', 'project:mobile:reference-browser:releng:signing:', 'project:mozilla:application-services:releng:signing'],
+            sign_chain_of_trust      => true,
+            verify_chain_of_trust    => true,
+            verify_cot_signature     => true,
+            cot_product              => 'application-services',
+            github_oauth_token       => secret('scriptworker_github_oauth_token_production'),
+            datadog_api_key          => secret('scriptworker_datadog_api_key'),
+            gpg_keyfile              => 'KEY_dep',
```
Comment (mitchhentges) [inline modules/signing_scriptworker/manifests/settings.pp]: Should this be `KEY_dep`?

File: modules/signing_scriptworker/templates/dep-passwords-appsv.json.erb (hunk 1 above)
Comment (mitchhentges) [reply modules/signing_scriptworker/templates/dep-passwords-appsv.json.erb]: I don't think we want the _focus_ dep key (is that what we're using for `android-components`?) but I don't think that we need an `application-services`-specific key. Can we use a generic "mozilla library" dep and release signing key?

File: modules/signing_scriptworker/templates/passwords-appsv.json.erb (hunk 2 above)
Comment (mitchhentges) [reply modules/signing_scriptworker/templates/passwords-appsv.json.erb]: Looks like this is the same key that's used by `android-components`, right? I think that re-using this key for other "mozilla library" signing should be fine.
Thinking about key management: downstream projects need to trust the signing key of the signed dependencies they use. It makes sense to have to manage different keys for different organizations, but if they need to add a new trusted signing key for each new _dependency_ (or group of dependencies) that they import, I'm not sure that that makes sense. Trusting a single "mozilla library" key makes the most sense, IMHO

File: modules/signing_scriptworker/manifests/settings.pp (hunk 3 above)
Comment (MihaiTabara) [reply modules/signing_scriptworker/manifests/settings.pp]: Good catch, I addressed it in a follow-up.

File: modules/signing_scriptworker/manifests/settings.pp (hunk 4 above)
Comment (MihaiTabara) [reply modules/signing_scriptworker/manifests/settings.pp]: Good catch, I addressed it in a follow-up.

File: modules/signing_scriptworker/templates/dep-passwords-appsv.json.erb (hunk 1 above)
Comment (MihaiTabara) [reply modules/signing_scriptworker/templates/dep-passwords-appsv.json.erb]: Good call, I pointed to the gpg default one,.

File: modules/signing_scriptworker/templates/passwords-appsv.json.erb (hunk 2 above)
Comment (MihaiTabara) [reply modules/signing_scriptworker/templates/passwords-appsv.json.erb]: Sounds good, let's keep this as is 👍

**Comment by MihaiTabara** [comment]
@mitchhentges ready for your re-review 😎

**Pull request merged**